### ISSUE TITLE
[Task] Change IP lookup service to provider that supplies SSL enabled endpoint.

### ISSUE THREAD

**Issue by deepseth** [Tautulli/Tautulli#264]
Running PlexPy behind an apache2 reverseproxy, access to apache2 interface is via HTTPS.

IP-API lookups are done at the client side rather than by the PlexPy app serverside, and HTTP requests from HTTPS pages are blocked by browser security:

[blocked] The page at https://myserver.com/ppy/history was not allowed to display insecure content from http://ip-api.com/json/1.2.3.4.

**Comment by drzoidberg33** [comment]
Yes, this is a known problem as ip-api doesn't supply an SSL end-point on it's free service.

You can bypass this on your browser however, see here: https://247support.custhelp.com/app/answers/detail/a_id/11736/~/chrome-and-firefox%3A-blocked-mixed-(https-and-http)-content

**Comment by deepseth** [comment]
Yeah I'm not sure disabling a security feature is the best option ;-)

**Comment by drzoidberg33** [comment]
They bypass only applies for the PlexPy page and doesn't affect other sites. The only data that will be sent over the non-SSL connection will be the ip address you are requesting.

There is unfortunately no other way currently to fix this while using ip-api.

Maybe somebody could take a look at changing the provider to one that does supply a free SSL end-point and submit a PR.

**Comment by JonnyWong16** [comment]
[Telize](http://www.telize.com) looks promising.

> There is no rate limit of any sort at the moment, and the service is free for everyone to use.
> BREAKING NEWS : Telize is now also available via HTTPS.

**Comment by Hellowlol** [comment]
telize looks awesome

**Comment by drzoidberg33** [comment]
Quick fix, I pushed a PR through. Please take a look and see if there is anything amiss.

**Comment by drzoidberg33** [comment]
Merged.

**Comment by Hellowlol** [comment]
Close this?

**Comment by drzoidberg33** [comment]
I'll close the completed issues once they're merged into master.

**Comment by realvz** [comment]
Bad news folks, Telize is not going to be free anymore. More info here: http://www.cambus.net/adventures-in-running-a-free-public-api/

**Comment by drzoidberg33** [comment]
Thanks for the info.

I might just host the Telize service on my own server.

**Comment by realvz** [comment]
http://dev.maxmind.com/geoip/legacy/geolite/

Here's the database Telize uses. If a service is not available, perhaps this can be used to gather information.

**Comment by drzoidberg33** [comment]
I've set up the Telize service on one of my own servers for the interim, until somebody can find something else or while it's not a headache to maintain.

**Comment by deepseth** [comment]
Will involve a bunch more work, but maybe move the GeoIP lookup logic away from the client-side, and onto the server-side app? That then gets around the issue of whether or not the API to lookup from is providing via HTTPS or not.

**Comment by drzoidberg33** [comment]
The telize service I've set up seems to be working fine. Closing this.
